tests/application/cms/test_views.py

```diff
@@ -978,3 +978,72 @@ def test_view_measure_version_by_measure_version_id_404_if_no_measure_version(te
     response = test_app_client.get(url_for("cms.view_measure_version_by_measure_version_id", measure_version_id=456))
 
     assert response.status_code == 404
+
+
+class _TestVisualisationBuilder:
+    ROUTE_NAME = None
+
+    def builder_url(self, measure_version):
+        if not self.ROUTE_NAME:
+            raise NotImplementedError
+
+        return url_for(
+            self.ROUTE_NAME,
+            topic_slug=measure_version.measure.subtopic.topic.slug,
+            subtopic_slug=measure_version.measure.subtopic.slug,
+            measure_slug=measure_version.measure.slug,
+            version=measure_version.version,
+            dimension_guid=measure_version.dimensions[0].guid,
+        )
+
+    def test_custom_classification_select_populates_with_known_classification(
+        self, test_app_client, logged_in_rdu_user
+    ):
+        measure_version = MeasureVersionWithDimensionFactory(version="1.0")
+        response = test_app_client.get(self.builder_url(measure_version))
+
+        assert response.status_code == 200
+        page = BeautifulSoup(response.data.decode("utf-8"), "html.parser")
+        classification_select = page.find("select", {"id": "custom_classification__selector"})
+        classification_options = classification_select.find_all("option")
+        assert len(classification_options) == 29
+        assert {option.get_text(strip=True) for option in classification_options} == {
+            "4 - Asian, Black, White, Other inc Mixed",
+            "4 - Black, Mixed, White, Other inc Asian",
+            "5 - Asian, Black, White British, White other, Other inc Mixed",
+            "6 - Asian, Black African, Black Caribbean, Black other, White, Other",
+            "6 - Asian, Black African, Black Caribbean, Black other, White, Other - " "Aggregates",
```

Comment:
Why do the "Aggregates" ones have the funky `" "` in the middle?

Reply:
Good question ... I don't know 🤔 I've removed them and it seems fine. I must've slipped up somewhere.